backend/src/main/java/ai/verta/modeldb/versioning/blob/container/DatasetContainer.java

```diff
@@ -254,12 +256,31 @@ public void processAttribute(
           RdbmsUtils.convertAttributesFromAttributeEntityList(
               this, ModelDBConstants.ATTRIBUTES, blobExpanded.getAttributesList());
       List<AttributeEntity> newAttributeEntities = new ArrayList<>();
+      List<AttributeEntity> existingAttributes =
+          VersioningUtils.getAttributeEntities(
+              session, repoId, commitHash, blobExpanded.getLocationList(), null);
       if (addAttribute) {
-        newAttributeEntities.addAll(newOrUpdatedAttributeEntities);
+        if (!existingAttributes.isEmpty()) {
+          for (AttributeEntity newAttribute : newOrUpdatedAttributeEntities) {
+            for (AttributeEntity existingAttribute : existingAttributes) {
+              if (existingAttribute.getKey().equals(newAttribute.getKey())) {
+                Status status =
+                    Status.newBuilder()
+                        .setCode(com.google.rpc.Code.ALREADY_EXISTS_VALUE)
+                        .setMessage(
+                            "Attribute being logged already exists. existing attribute Key : "
+                                + newAttribute.getKey())
+                        .build();
+                throw StatusProto.toStatusRuntimeException(status);
+              } else {
+                newAttributeEntities.add(newAttribute);
```

Comment:
this should be outside the inner for

Reply:
Both ways will work. I am adding it to the new list when the key is not matched with the existing key. but your suggestion also works like: `newAttributeEntities.addAll(newOrUpdatedAttributeEntities);`

Reply:
Resolved with a better way